docs/sphinx/source/index.rst

```diff
@@ -60,11 +60,54 @@ Contents
    :maxdepth: 2
    
    self
+   package_overview
    whatsnew
+   usage
+   modules
+   classes
    comparison_pvlib_matlab
    variables_style_rules
-   pvlib
 
+   
+Contents (potential new ToC)
+============================
+
+* What’s New
+* Overview: Why pvlib?
+    * Introduction
+    * Modeling paradigms
+    * User extensions
+    * What it is not
+    * Comparison with PVLIB_MATLAB
+* Installation
+* Examples
```

Comment:
I suggest to keep the top level.
This helps to reduce `maxdepth` to 1 once there are many sub-pages.